### PR TITLE
Update samples

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
* Use https://eclipse-che.github.io/che-plugin-registry/ for plugin devfiles instead of https://che-plugin-registry-main.surge.sh
* Use che-code as editor plugin for all samples except explicit Theia sample
* Fix issues with commands (wrong path -- use `$PROJECT_SOURCE` instead of `$PROJECTS_ROOT`)
* Add `dev` containers to main samples (`code-latest.yaml` and `theia-latest.yaml`) so that container contributions are used (editor should merge component into dev component)
* Use contributions for all plugins (rather than plugin components)

### What issues does this PR fix or reference?
N/A

### Is it tested? How?
Test that samples generally work as expected. If testing on minikube, it may be necessary to pre-pull images, as the UDI image is _very_ large (3.7GB):

```bash
minikube ssh docker pull quay.io/devfile/universal-developer-image:latest
# Or, depending on the sample used
minikube ssh docker pull quay.io/devfile/universal-developer-image@sha256:d1709bbdfa076474f58f796026a2ed2dd3b24fea7e51ce2cc984e729663ff62c
```

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

**Review by AObuchow — COMMENTED** [review]
Tested on OpenShift 4.12.

- `pod-overrides` sample failed since the kata RuntimeClass was not found (to be expected) 
- `theia-latest` sample works as expected
- Unfortunately, for all the samples using Che Code, I got the following error about the application not being served at the endpoint:

image

**Comment by amisevsk** [comment]
> Unfortunately, for all the samples using Che Code, I got the following error about the application not being served at the endpoint

Hm, I'll have to take a look

**Comment by l0rd** [comment]
I have tested on a fresh OCP 4.12 cluster (I have installed DWO with `make install`) and `k apply -f code-latest.yaml` failed with "Init Container project-clone has state CrashLoopBackOff" and from the logs:

> 2023/01/31 11:44:54 Failed to get temporary directory for setting up projects: mkdir /projects/project-clone-1091890368: permission denied

**Comment by l0rd** [comment]
Otherwise I have the same result as @AObuchow but [removing `path: '?tkn=eclipse-che'` in vscode definition](https://gist.githubusercontent.com/l0rd/b6045738e5158231bf780bd19cd0ddce/raw/49664affd70366f7a76f1d6b2340800f8d3d5791/vscode-withouth-path.yaml) (or opening the main URL without `?tkn=eclipse-che`) fixes the problem. But that's not related to this PR so I have opened a [separate one](https://github.com/eclipse-che/che-plugin-registry/pull/1562).

**Comment by nickboldt** [comment]
related to https://github.com/eclipse/che/issues/21973

Note that the `?tkn=eclipse-che` fix has also been done downstream for 3.5/3.x: https://github.com/redhat-developer/devspaces/pull/880

**Comment by AObuchow** [comment]
> Note that the `?tkn=eclipse-che` fix has also been done downstream for 3.5/3.x: [redhat-developer/devspaces#880](https://github.com/redhat-developer/devspaces/pull/880)

Will do a re-test, thanks Nick :+1: :sunglasses:

**Comment by AObuchow** [comment]
I've opened the following 2 issues which are related to this PR. They could be solved in separate commits in this PR, or as separate PRs:

https://github.com/devfile/devworkspace-operator/issues/1028
https://github.com/devfile/devworkspace-operator/issues/1027

**Comment by AObuchow** [comment]
I seem to be encountering some CheCode (?) bugs:

When trying out the git-clone sample, code latest and the ephemeral storage sample, I got the following errors on Firefox specifically:
image
image

Refresh the page would cause either of these errors would pop up.

On the per-workspace sample, the editor initially started up, but clicking on the web-nodejs-sample project in the file explorer (which was written in yellow text) gave the following error:

image


This issue occured on Chrome less often, but still happened sometimes (`The workbench failed to connect to the server (Error: WebSocket close with status code 10006)`).

Another oddity: Though the git-clone sample workspace opened in Chrome, there's no `devworkspace-operator` project in the file explorer? And the devfile in the workspace is incorrect?

image

exec'ing into the workspace pod (or opening up a terminal in the editor) showed that the `devworkspace-operator` project was there. I thought at first this issue occured cause I had multiple workspaces started at the same time, but this still happened after deleting all workspaces and starting the git-clone sample again.

I'm assuming these are all Che Code bugs but not sure yet. Will file a bug report about these issues. 
Even then, it might be best for these issues to be resolved before we merge this PR so that the samples all work as expected.

**Comment by amisevsk** [comment]
Retested samples on OpenShift 4.12, using Chromium 109.0.5414.74 and Firefox 109.0:

* Che-code samples start now, after Mario's fix. This led me to discover a bug in DWO's 'basic' routing reconciler: https://github.com/devfile/devworkspace-operator/issues/1031. I'll fix this one asap.
* I cannot reproduce issues around lost connection, etc., and workspaces work as expected (to my experience). I'm able to open a terminal session, files, etc.
* The issue for the git clone sample is expected behavior; when Code launches in the workspace, it opens the first project rather than the `projects` folder. You can open a workspace with all your projects manually.
    * This shouldn't be a problem, as devfiles are currently intended to only have one project

**Comment by amisevsk** [comment]
@AObuchow This PR is still waiting on approve/request changes

**Review by AObuchow — APPROVED** [review]
Tried all the updated samples and everything works as expected :pray:

**Pull request merged**
